### DISCUSSION TITLE
Wrong line number on error messages when auto-imports are used

### DISCUSSION THREAD

**Discussion by Heulitig** [fastn-stack/fastn#335] [Ideas & RFCs]
<details>
<summary>Issue Created</summary>

https://github.com/FifthTry/ftd/issues/230

</details>

Currently, when we use `auto-imports`, a `prefixed document` is generated before passing it to ftd. `Ftd` receives this prefixed document instead of the actual document and since the actual document contents are shifted down that's why we are getting wrong line number on the error messages.
 
```markdown
Prefixed Body: {Auto-imports}\n\n{Actual document}  
Shift in line number = (no of auto-imports used + 2)
```
Need some way to fix this line number issue by sending this data regarding auto-imports separately and avoid altering the original line numbers of the document.

# How To Solve This?

Simplest solution would be to pass both document source and `offset` when we are continuing after [continue_after_import ](https://github.com/FifthTry/ftd/blob/main/src/p2/interpreter.rs#L577).

```rust
fn continue_after_import(mut self, id: &str, source: &str, offset: usize) -> ftd::p1::Result<Interpreter>
```

Then whenever ftd is printing, it will print `line-number - offset` as line number. 

Problem with this approach: What if file corresponding to auto import does not exist or has an issue during import?

**Comment by Heulitig** [comment]
What if we use some kind of line seperator to denote the end of the auto-imports in the prefixed document like this

```markdown
[AUTO-IMPORTS]
;; <EO-AI>
[ORIGINAL-DOCUMENT]
```
And start computing the line numbers after this `<EO-AI>` (End of auto-imports). And before `<EO-AI>`, the line numbers will be undefined because those are not part of the actual document or maybe we can give some default line number maybe `-1` to denote that these lines are from auto-imports in case we want to display those in error msgs which might originate from these auto-imports.

**Comment by amitu** [comment]
@Heulitig this is not a discussion material. What you are describing is a bug. Bugs should be filed as issues. Discussion is purely for discussion of future developments, new features etc. 

I have written about this in [Issues vs Discussions thread](https://github.com/orgs/FifthTry/discussions/69).
